### PR TITLE
Pipeline/feature extractor

### PR DESCRIPTION
**Pull request by mariaogryczak**
Model for feature extractor pre-training.
I don't think that we should push it to main, as it is just a tool for pre-training, however, it is good to have it on github.
Until we have data, we can not use it.

### CODE REVIEW

**Comment by AgaMiko** [comment]
> It is highly probable, that this PR will not pass the check.

@mariaogryczak the error says: "FileNotFoundError: [Errno 2] No such file or directory: 'assets/sanity_check_data/test_hamnosys2.txt'" so maybe switching the path in the `annotation_file` will be ok for now

**Comment by mariaogryczak** [comment]
There is exactly a substitution of model by model_for_pretraining and additional changes in train, dataset and classification_mode to make it work, but the last three should be

**Comment by mariaogryczak** [comment]
In that case, I will add a parameter and do a PR to main :)

File: train.py [hunk 1]
```diff
@@ -9,7 +9,7 @@
 from torch.utils.data import DataLoader, random_split
 
 from datasets.dataset import ImglistToTensor, PadCollate, VideoFrameDataset
-from models.model import GlossTranslationModel
+from models.model_for_pretraining import GlossTranslationModel
```
Comment (AgaMiko) [inline train.py:12]: I would add it in readme but remove from train :)

**Comment by mariaogryczak** [comment]
Option of pre-training was added in parser that allows using a model_for_pretraining.py file instead of model.py. 
Additionally, all model parameters were gathered into MODEL_CONFIG dict to prevent parameters duplication for each possible model.

**Review by AgaMiko — APPROVED** [review]

File: datasets/dataset.py [hunk 2]
```diff
@@ -117,6 +117,7 @@ def __init__(
         root_path: str,
         annotationfile_path: str,
         classification_mode: str,
+        pre_training: bool,
```
Comment (wiktorfilipiuk) [inline datasets/dataset.py:120]: I suggest using convention is_pretraining as bool variable names

File: models/model_for_pretraining.py [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+        capture_stderr=False,
+    )
+
+
+class PreTrainingModel(pl.LightningModule):
+    """Awesome model for Gloss Translation"""
+
+    def __init__(
+        self,
+        MODEL_CONFIG = {"lr": 1e-5,
```
Comment (wiktorfilipiuk) [inline models/model_for_pretraining.py:32]: let's not name constructor params with capital letters :)

Comment (majsylw) [reply models/model_for_pretraining.py:32]: @wiktorfilipiuk can we easly combine this to read parameters from file?

**Review by majsylw — APPROVED** [review]

File: models/model_for_pretraining.py (hunk 3 above)
Comment (AgaMiko) [reply models/model_for_pretraining.py:32]: It would be great!

Comment (mariaogryczak) [reply models/model_for_pretraining.py]: I would do 'reading from file'
 in separate PR. It is not strictly connected with this one and there are a lot of parameters that are now pushed by parser.

**Pull request merged**